tests/Platforms/DB2PlatformTest.php

```diff
@@ -269,10 +269,110 @@ public function testGeneratesColumnTypesDeclarationSQL(): void
             $this->platform->getDateTimeTypeDeclarationSQL(['version' => true]),
         );
 
-        self::assertEquals('TIMESTAMP(0)', $this->platform->getDateTimeTypeDeclarationSQL($fullColumnDef));
+        self::assertEquals('TIMESTAMP(10)', $this->platform->getDateTimeTypeDeclarationSQL($fullColumnDef));
         self::assertEquals('TIME', $this->platform->getTimeTypeDeclarationSQL($fullColumnDef));
     }
 
+    public function testGeneratesTypeDeclarationForDatetimes(): void
```

Comment:
I would deliberately not introduce unit tests for SQL. They produce code coverage but don't test interaction with the database. It's better to make sure that all the new behaviors are tested by integration tests.

Reply:
I deliberately added them to ensure that where no length (or a zero length) is provided, the outcome is the SQL declaration that is currently generated. For instance the current defaults are DATETIME2(6) and TIME(0) on SQL server but DATETIME or TIMESTAMP and TIME on MySQL. The MySQL versions would be equally valid on SQL server but result in a migration if they were used. So I do think they add something (and they picked up an error when updating the patch set). 

If you still think they are useless, I can delete them.